### PR TITLE
Add timeout knobs

### PR DESCRIPTION
**Pull request by kakkoyun**
This adds all the possible knobs to be more flexible for the adjustments.
Also changes the default values.

### CODE REVIEW

**Comment by metalmatze** [comment]
Do we really want to expose all of these via flags? :thinking: 
Sane defaults should be good enough for 90% of these, don't you think?

**Comment by kakkoyun** [comment]
@metalmatze Yes, happy to remove flags 👍 We have too many of them.

File: internal/proxy/proxy.go [hunk 1]
```diff
@@ -23,6 +24,18 @@ const (
 	DefaultBufferSizeBytes = 32 * 1024
 	// DefaultFlushInterval TODO
 	DefaultFlushInterval = time.Duration(-1)
+	// DefaultTimeout TODO
```
Comment (squat) [inline internal/proxy/proxy.go:27]: What is left to do here? Simply right the comment?

File: internal/proxy/proxy.go [hunk 2]
```diff
@@ -23,6 +24,18 @@ const (
...
+	DefaultTimeout = 30 * time.Second
```
Comment (squat) [inline internal/proxy/proxy.go:28]: It seems that most of these consts exposed as public identifiers are never used outside of this package. Can we make them unexported if they are not used?

File: internal/proxy/proxy.go (hunk 1 above)
Comment (kakkoyun) [reply internal/proxy/proxy.go:27]: yes :)

File: internal/proxy/proxy.go [hunk 3]
```diff
@@ -17,12 +18,26 @@ import (
 )
 
 const (
-	// DefaultBufferCount TODO
+	// DefaultBufferCount The default value for the maximum size of the buffer pool for the reverse proxy.
```
Comment (squat) [inline internal/proxy/proxy.go:21]: nit: s/The/is the/

File: internal/server/server.go [hunk 4]
```diff
@@ -17,10 +17,19 @@ import (
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
+// DefaultGracePeriod The default value of the duration gracefully shuts down the server without interrupting any active connections.
```
Comment (squat) [inline internal/server/server.go:20]: same for these consts

**Comment by squat** [comment]
let's squash the commit :)

**Pull request merged**